src/components/TitleBar/ViewTitleBar.js

```diff
@@ -64,19 +71,38 @@ const ViewTitleBar = props => {
         ? i18n.t('Hide description')
         : i18n.t('Show description')
 
-    const showHideDescription = () => {
-        onShowHideDescription()
-        toggleMoreOptions()
+    const onToggleShowDescription = () => {
+        apiPostShowDescription(!showDescription)
+            .then(() => {
+                updateShowDescription(!showDescription)
+                toggleMoreOptions()
+            })
+            .catch(() => {
+                const msg = showDescription
+                    ? i18n.t('Failed to hide description')
+                    : i18n.t('Failed to show description')
+                warningAlert.show({ msg })
+            })
     }
 
     const toggleStarredDashboardLabel = starred
         ? i18n.t('Unstar dashboard')
         : i18n.t('Star dashboard')
 
-    const toggleStarredDashboard = () => {
-        // TODO use dataQuery directly here with create/delete mutations ?!
```

Comment:
My understanding of dataEngine usage is that if a query/mutation is only needed in one place, it should be as close as possible to where it's used.
Starring/unstarring only happens in this component and the mutation is not needed anywhere else, so I was wondering if it should be used here directly with the `useMutation` hook instead of define it in a separate file.
That should simplify the code a bit.
@amcgee can comment further here.

Some relevant discussions:
https://dhis2.slack.com/archives/G3TL0J145/p1602848442010400
https://dhis2.slack.com/archives/CLJSVA1RV/p1602144467062200
https://dhis2.slack.com/archives/CPGUFVC56/p1605263408418500

Reply:
That's fine with me.

One goal I have is to dynamically create Cypress api tests from the data query/mutations in the app. I haven't thought a lot about how to do that, but one thing I had in mind was to use convention by keeping all the query/mutation definitions in a standard location (e.g. `src/api`). But that shouldn't conflict with the idea of using the `useMutation` hook.

Reply:
Updated to use the `useDataMutation` hook.

Reply:
Reverted to the original solution because of console warnings and failing cypress test

Reply:
👍